tests/rag_answers/custom_deepeval/metrics/test_factual_correctness.py

```diff
@@ -0,0 +1,415 @@
+            else:
+                assert metric.reason == expected_reason  # type: ignore
+
+        @pytest.mark.asyncio
+        @pytest.mark.parametrize(
+            "strict_mode, threshold, TP, FP, expected_score, expected_success",
+            [
+                (
+                    False,
+                    0.7,
+                    ["fact1", "fact2"],
+                    ["fact3"],
+                    0.6667,
+                    False,
+                ),  # regular mode, return raw score
+                (False, 0.7, ["fact1", "fact2"], [], 1.0, True),
+                (
+                    True,
+                    0.7,
+                    ["fact1", "fact2", "fact3"],
+                    ["fact4"],
+                    0.0,
+                    False,
+                ),  # strict: score < strict-threshold of 1
+                (
+                    True,
+                    0.7,
+                    ["fact1", "fact2"],
+                    [],
+                    1.0,
+                    True,
+                ),  # strict: score == strict-threshold of 1
+                (
+                    True,
+                    0.7,
+                    ["fact1"],
+                    ["fact2", "fact3"],
+                    0.0,
+                    False,
+                ),  # strict: score <= threshold so should not trigger strict mode logic
+            ],
+        )
+        async def test_strict_mode_can_be_configured(
+            self,
+            strict_mode: bool,
+            threshold: float,
+            TP: list,
+            FP: list,
+            expected_score: float,
+            expected_success: bool,
+            mock_native_model: Mock,
+            test_case: LLMTestCase,
+        ):
+            classified_facts = ClassifiedFacts(TP=TP, FP=FP, FN=[])
+
+            metric = FactualCorrectnessMetric(
+                model=mock_native_model,
+                threshold=threshold,
+                strict_mode=strict_mode,
+            )
+
+            metric._a_classify_statements = AsyncMock(return_value=classified_facts)
+
+            # let _calculate_score and _finalise_evaluation run naturally
+            result = await metric.a_measure(test_case)
+
+            assert round(result, 4) == round(expected_score, 4)
+            assert metric.success == (expected_success)
+
+        @pytest.mark.asyncio
+        async def test_native_model_sets_evaluation_costs(
+            self,
+            mock_native_model: Mock,
+            test_case: LLMTestCase,
+            fact_classification_result: FactClassificationResult,
+        ):
+            expected_cost = 0.05
+
+            mock_native_model.a_generate = AsyncMock(
+                return_value=(fact_classification_result, expected_cost)
+            )
+
+            metric = FactualCorrectnessMetric(model=mock_native_model)
+            metric.using_native_model = True  # ensure we hit native logic
```

Comment:
We shouldn't need this line as it's internal details and could potentially conceal a bug. We don't expect the get costs without a native model so we'd expect the test to fail to help reveal the issue.